miniwallet/config.ts

```diff
@@ -0,0 +1,146 @@
+      initialOperators: JSON.parse(process.env.MINIWALLET_INITIAL_OPERATORS || '[]'),
+      initialUserLimit: ethers.utils.parseEther(process.env.MINIWALLET_INIITIAL_USER_LIMIT || '1000000'),
+      initialAuthLimit: ethers.utils.parseEther(process.env.MINIWALLET_INIITIAL_AUTH_LIMIT || '100000')
+    },
+    miniID: {
+      baseUri: process.env.MINIID_BASE_URI || 'ipfs://Qmc8DVEthq7cZMTMyZ2NQ8dHkG99n549DMBwNzAypQgXe1/MiniID'
+    },
+    mini721: {
+      saleIsActive: process.env.MINI721_DEPLOY_SALES_IS_ACTIVE || false,
+      metadataFrozen: process.env.MINI721_DEPLOY_METADATA_FROZEN || false,
+      provenanceFrozen: process.env.MINI721_DEPLOY_PROVENANCE_FROZEN || false,
+      max721Tokens: process.env.MINI721_DEPLOY_MAX721_TOKENS || 1000000000000,
+      mintPrice: ethers.utils.parseEther(process.env.MINI721_DEPLOY_MINT_PRICE || '0'),
+      maxPerMint: process.env.MINI721_DEPLOY_MAX_PER_MINT || 1,
+      baseUri: process.env.MINI721_DEPLOY_BASE_URI || 'ipfs://QmPcY4yVQu4J2z3ztHWziWkoUEugpzdfftbGH8xD49DvRx/',
+      contractUri: process.env.MINI721_DEPLOY_CONTRACT_URI || 'ipfs://Qmezo5wDKz7kH'
+    },
+    mini1155: {
+      deploy: {
+        name: process.env.MINI1155_DEPLOY_NAME || 'MiniWallet',
+        symbol: process.env.MINI1155_DEPLOY_SYMBOL || 'Mini1155',
+        salt: ethers.utils.formatBytes32String(process.env.MINI1155_DEPLOY_SALT || '1'),
+        baseUri: process.env.MINI1155_DEPLOY_BASE_URI || 'ipfs://QmPcY4yVQu4J2z3ztHWziWkoUEugpzdfftbGH8xD49DvRx/',
+        contractUri: process.env.MINI1155_DEPLOY_CONTRACT_URI || 'ipfs://QmdKB6d1zT7R8dNmEQzc6N1m5p2LDJZ66Hzu8F4fGhdVrq',
+        revenueAccount: process.env.MINI1155_DEPLOY_REVENUE_ACCOUNT,
+        saleIsActive: process.env.MINI1155_DEPLOY_SALE_IS_ACTIVE === 'true' || process.env.TEST_MINI1155_DEPLOY_SALE_IS_ACTIVE === '1',
+        metadataFrozen: process.env.MINI1155_DEPLOY_METADATA_FROZEN === 'true' || process.env.TEST_MINI1155_DEPLOY_METADATA_FROZEN === '1',
+        mintPrice: ethers.utils.parseEther(process.env.MINI1155_DEPLOY_MINT_PRICE || '0'),
+        exchangeRatio: process.env.MINI1155_DEPLOY_EXCHANGE_RATIO || 0, // will be 20 after sale
+        rareProbabilityPercentage: process.env.MINI1155_DEPLOY_RARE_PROBABLITY_PERCENTAGE || 1,
+        maxPerMint: process.env.MINI1155_DEPLOY_MAX_PER_MINT || 10,
+        // standard token configuration
+        s: {
+          tokenId: process.env.MINI1155_DEPLOY_STANDARD_TOKEN_ID || 1,
+          maxSupply: process.env.MINI1155_DEPLOY_STANDARD_TOKEN_MAX_SUPPLY || 770,
+          personalCap: process.env.MINI1155_DEPLOY_STANDARD_TOKEN_PERSONAL_CAP || 10
+        },
+        // rare token configuration
+        r: {
+          tokenId: process.env.MINI1155_DEPLOY_RARE_TOKEN_ID || 2,
+          maxSupply: process.env.MINI1155_DEPLOY_RARE_TOKEN_MAX_SUPPLY || 7, // will be 84 after Sale
+          personalCap: process.env.MINI1155_DEPLOY_RARE_TOKEN_PERSONAL_CAP || 1
+        }
+      }
+    }
+  },
+
+  test: {
```

Comment:
test can move to its own config file

Reply:
Refactored approach and commit are [here](https://github.com/polymorpher/sms-wallet/issues/13#issuecomment-1246191757)